### ISSUE TITLE
Upgrading to v5.1.0 jqm ui-icon-calendar below date picker

### ISSUE THREAD

**Issue by gregh3269** [jtsage/jtsage-datebox#469]
Something that I have noticed is the calendar icon is below the date picker field.  If I then do a manual page refresh it aligns correctly and it alters the html code.

incorrect alignment before manual refresh:
```
<div id="visitDate5jH12" class="ui-input-has-clear">
<div class="ui-input-text ui-body-inherit ui-corner-all ui-shadow-inset">
<input name="visitDate" id="visitDateString5jH12" type="text" value="20/05/19" data-role="datebox" data-options="{"mode": "calbox", "theme_headerTheme" : "b", "useFocus":true, "overrideDateFormat": "%d/%m/%y", "minDays" : 30.0, "maxDays" : 0.0, "disabledState": "disabled", "displayDropdownPosition":"centerMiddle" }" readonly="readonly">
</div>
<a href="javascript: return false;" class="dbOpenButton ui-input-clear ui-btn ui-icon-calendar ui-btn-icon-notext ui-corner-all" title="Open Date Picker">Open Date Picker</a>
</div>
```
correct alignment, after manual refresh:
```
<div id="visitDateuCuZY">
<div class="ui-input-text ui-body-inherit ui-corner-all ui-shadow-inset ui-input-has-clear"><input name="visitDate" id="visitDateStringuCuZY" type="text" value="20/05/19" data-role="datebox" data-options="{"mode": "calbox", "theme_headerTheme" : "b", "useFocus":true, "overrideDateFormat": "%d/%m/%y", "minDays" : 30.0, "maxDays" : 0.0, "disabledState": "disabled", "displayDropdownPosition":"centerMiddle" }" readonly="readonly">
<a href="javascript: return false;" class="dbOpenButton ui-input-clear ui-btn ui-icon-calendar ui-btn-icon-notext ui-corner-all" title="Open Date Picker">Open Date Picker</a></div>
</div>

```
The manual refresh, moves the position of the icon up one div.

**Comment by gregh3269** [comment]
I guess jqm adds the enclosing div (ui-input-text ui-body-inherit ui-corner-all ui-shadow-inset) after so the .parent() targets the correct element?
In my tests this works, adds the auto div stuff then w.style_inWrap targets the correct .parent()
```
w.d.input.parent().enhanceWithin();
w.d.wrap = w.style_inWrap(w.d.input, o.theme_openButton);
```

**Comment by jtsage** [comment]
Gross.  thought I got all these.  Just so we are absolutely working from the same place, can you send along that bit in your original source (the bit from before either datebox of jqm enhance it).

99% of the time, it's because I've marked it up slightly differently in my test version, but it should be able to detect all of the possible valid ways, and it's clearly missing one.

i.e. 

```html

<div class="ui-field-contain">
     <label for="db">yep</label>
     <input name="db" type="text" data-mini="true" data-role="datebox" id="db" />
</div>
```

(of course, I already notice my sample is not actually in a &lt;form> tag, just the .ui-content page div.  Also, are you doing multiple pages per file?  Shouldn't matter, but give me a good place to start)

Thanks!

~j

**Comment by gregh3269** [comment]
From the source
```
<div id="visitDate${uuId}" class="ui-select">
	<input name="visitDateString" id="visitDateString${uuId}" type="text" value="${visitDateString}" data-role="datebox" data-options='{"mode": "calbox", "theme_headerTheme" : "b", "useFocus":true, "overrideDateFormat": "${format}", "minDays" : ${mindays}, "maxDays" : ${maxdays}, "disabledState": "disabled", "displayDropdownPosition":"centerMiddle" }' readonly="readonly" data-inline="true" />
</div>

```

**Comment by jtsage** [comment]
Looks like this will do it:

```js
style_inWrap = function ( originalInput ) {
	originalInput.parent().enhanceWithin();
	return originalInput.parent().addClass("ui-input-has-clear");
};
```

Only reason I didn't use your's outright, is the style_inWrap function is defined per css framework, and that other bit of code exists in every framework - while it would work great for jQM, it'll fail everywhere else.

**Comment by gregh3269** [comment]
That works ok (I had originally posted it like but edited the comment #2).

**Comment by jtsage** [comment]
Fix pushed to release, SHA: 366e8c237edd874964ed1f717df8abb2a8a1a3ee (5.1.2)

Closing issue, if this comes back, or I missed something, feel free to re-open.
